src/shogun/machine/Composite.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ */
+#ifndef __COMPOSITE_H
+#define __COMPOSITE_H
+#include <memory>
+#include <shogun/ensemble/CombinationRule.h>
+#include <shogun/features/Features.h>
+#include <shogun/labels/Labels.h>
+#include <shogun/machine/EnsembleMachine.h>
+#include <shogun/machine/Machine.h>
+#include <shogun/transformer/Transformer.h>
+#include <vector>
+#include <variant>
+#include <shogun/base/variant.h>
+
+template<class... Ts> struct overloaded : Ts... { using Ts::operator()...; };
+template<class... Ts> overloaded(Ts...) -> overloaded<Ts...>;
```

Comment:
Isn't this defined anywhere else? If not you should move this to another file, because it should be pretty common